onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -124,79 +123,28 @@ void AddBinaryOperator(int32_t op_type,
   model_builder.AddOperation(op_type, input_indices, {output}, {output_operand_type}, {output_is_nhwc});
 }
 
-int GetType(const ONNX_NAMESPACE::ModelProto& model_proto,
-            const std::string& name) {
-  int invalid_type = ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED;
-  for (const auto& input : model_proto.graph().input()) {
-    if (input.name() != name)
-      continue;
-
-    return input.type().tensor_type().elem_type();
-  }
-
-  for (const auto& value_info : model_proto.graph().value_info()) {
-    if (value_info.name() != name)
-      continue;
-
-    if (!value_info.has_type()) {
-      return invalid_type;
-    } else if (!value_info.type().has_tensor_type()) {
-      return invalid_type;
-    }
-
-    return value_info.type().tensor_type().elem_type();
+bool GetType(const onnxruntime::Node& node, int32_t& type) {
+  type = ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED;
+  auto* type_proto = node.InputDefs()[0]->TypeAsProto();
+  if (!type_proto || !type_proto->tensor_type().has_elem_type()) {
```

Comment:
Unexpected that a GetType function given a node would default to only checking the first input type.